test/mpi/datatype/testlist

```diff
@@ -0,0 +1,65 @@
+contents 1
```

Comment:
It looks like you committed a new file - `testlist`, but this directory of the testsuite uses `testlist.in`. We should change this to only modify `testlist.in` to add the new test.